### PR TITLE
WebCmdlets fix issue with -Body and no -ContentType

### LINKED ISSUE #23827: Invoke-WebRequest and Invoke-RestMethod in Powershell 7.4.2 - errors using PUT method
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://learn.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

I'm receiving error calling WebCmdlets (Invoke-WebRequest, Invoke-RestMethod) using PUT method if I don't explicitly set content-type header.

Calling
`Invoke-WebRequest -Method PUT -Uri $someUri -Body $someBody`
Returns Error in PowerShell 7.4.2:
`Invoke-WebRequest: The given key 'Content-Type' was not present in the dictionary.`

(the error is thrown before any network call is even started)

The exact same call runs OK with PowerShell 7.4.1 and all lower versions that I was able to test.

Running: 
`Invoke-WebRequest -Method PUT -Uri $someUri -Body $someBody -ContentType 'application/x-www-form-urlencoded'`
also works OK - even with powershell 7.4.2.

I see the change in v7.4.2 release notes https://github.com/PowerShell/PowerShell/pull/21049
But, as I understand it, setting content-type (by request headers collection or -ContentType parameter) isn't required and omitting it shouldn't be any issue regardless of pwsh version. Content-type can be default ('application/x-www-form-urlencoded') or none depending on pwsh version.

According to the available docs:
https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/invoke-restmethod?view=powershell-7.4#-contenttype

> If this parameter is omitted and the request method is POST, Invoke-RestMethod sets the content type to application/x-www-form-urlencoded. Otherwise, the content type isn't specified in the call.

I could successfully call PUT requests without explicitly specifying Content-type just until pwsh 7.4.2.

### Expected behavior

```console
Invoke-WebRequest -Method PUT -Uri "http://fake" -Body "fake"
Invoke-WebRequest: Resource temporarily unavailable
```

### Actual behavior

```console
> Invoke-WebRequest -Method PUT -Uri "http://fake" -Body "fake"
Invoke-WebRequest: The given key 'Content-Type' was not present in the dictionary.
```

### Error details

```console
Invoke-WebRequest: The given key 'Content-Type' was not present in the dictionary.
```

### Environment data

```powershell
Name                           Value
----                           -----
PSVersion                      7.4.2
PSEdition                      Core
GitCommitId                    7.4.2
OS                             Ubuntu 22.04.4 LTS
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Visuals

_No response_

### PR DESCRIPTION
**Pull request by CarloToso**
# PR Summary
I reproduced the issue with every method except POST; now it should work correctly
In PS 7.4 and 7.4.1 PUT had the same behaviour as POST, so it worked without issues
Added some tests (checked with curl)

## PR Context
Fixes #23827
@iSazonov 

## PR Checklist

- [ ] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
  - Use the present tense and imperative mood when describing your changes
- [ ] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
  - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
  - [ ] None
  - **OR**
  - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.5/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
    - [ ] Experimental feature name(s): 
- **User-facing changes**
  - [ ] Not Applicable
  - **OR**
  - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
  - [ ] N/A or can only be tested interactively
  - **OR**
  - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
  - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
  - **OR**
  - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
    - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
      - [ ] Issue filed: 
    - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
      - [ ] Issue filed: 
    - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
      - [ ] Issue filed: 
    - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
      - [ ] Issue filed:

### CODE REVIEW

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs:1579]: It seems we never check the value for null. So we should care:
```suggestion
            if (WebSession.ContentHeaders.TryGetValue(HttpKnownHeaderNames.ContentType, out string contentType) && contentType is not null)
```

**Comment by iSazonov** [comment]
@borland  @bondido Could you please download compiled artifact from the PR and confirm that the issue is solved?

File: test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1:2799]: Please remove extra empty line.
```suggestion
```

**Comment by borland** [comment]
> @borland @bondido Could you please download compiled artifact from the PR and confirm that the issue is solved?

@iSazonov I'm not familiar with the CI setup here. I clicked through to Azure DevOps and found this:
https://dev.azure.com/powershell/PowerShell/_build/results?buildId=148781&view=artifacts&pathAsName=false&type=publishedArtifacts

I downloaded it, launched, and it said it was `PowerShell 7.5.0-preview.4` so I assume that is the correct thing.

From there, I ran the exact same test. On PowerShell 7.4.2 (windows x64) I get

```
Invoke-RestMethod: The given key 'Content-Type' was not present in the dictionary.
```
I do not see any entries on my local test server; PowerShell never sends the request.

On the 7.5.0-preview.4 pull request build, I do not get the error. The logs on my local server confirm that PowerShell is sending the request and the behaviour looks similar to older versions of PowerShell where this worked. Thank you!

For reference: My repro is fairly simple: I have a local aspnetcore webserver (but anything will do) and I'm sending a JSON PUT as follows:

```powershell
$header = @{}
$body = '{ }' // actual body content is specific to my webserver, but is irrelevant for the repro
Invoke-RestMethod -Method Put -Uri "<local server URI>" -Body $body -Headers $header
```

**Review by iSazonov — APPROVED** [review]

**Review by daxian-dbw — APPROVED** [review]

**Comment by adityapatwardhan** [comment]
We will wait for a preview release with this change before backporting to 7.4

**Comment by TravisEz13** [comment]
@PowerShell/powershell-maintainers 
Because this is a regression after 7.4 release, we believe this meets the bar for backport for 7.4

**Review by moazzem-pixel — COMMENTED** [review]
I need support

CarloToso:ContentType-errors

**Pull request merged**
